.github/workflow_scripts/tag-main.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env sh
+
+GITHUB_REPO="osmlab/atlas"
+RELEASE_BRANCH=main
+
+FUNCTION_NAME="tag-$RELEASE_BRANCH"
+CURRENT_BRANCH=$(echo $GITHUB_REF | awk 'BEGIN { FS = "/" } ; { print $3 }')
+
+echo "$FUNCTION_NAME: $GITHUB_REPO"
+echo "$FUNCTION_NAME: CURRENT_BRANCH = $CURRENT_BRANCH"
+echo "$FUNCTION_NAME: GITHUB_HEAD_REF = $GITHUB_HEAD_REF"
+
+if [ "$CURRENT_BRANCH" != "$RELEASE_BRANCH" ];
+then
+    echo "$FUNCTION_NAME: Exiting! Branch is not $RELEASE_BRANCH: ($CURRENT_BRANCH)"
+    exit 0;
+fi
+
+if [ -n "$GITHUB_HEAD_REF" ];
+then
+    PULL_REQUEST_NUMBER=$(echo $GITHUB_REF | awk 'BEGIN { FS = "/" } ; { print $3 }')
+    echo "$FUNCTION_NAME: Exiting! This is a Pull Request: $PULL_REQUEST_NUMBER"
+    exit 0;
+fi
+
+if [ "$MANUAL_RELEASE_TRIGGERED" != "true" ];
+then
+    echo "$FUNCTION_NAME: Exiting! This is not a release build."
+    exit 0;
+fi
+
+: ${MERGE_TAG_GITHUB_SECRET_TOKEN:?"MERGE_TAG_GITHUB_SECRET_TOKEN needs to be set in the workflow yml file!"}
+: ${GITHUB_SHA:?"GITHUB_SHA needs to be available to tag the right commit!"}
+
+export GIT_COMMITTER_EMAIL="github-actions@github.com"
+export GIT_COMMITTER_NAME="Github Actions CI"
+
+TEMPORARY_REPOSITORY=$(mktemp -d)
+git clone "https://github.com/$GITHUB_REPO" "$TEMPORARY_REPOSITORY"
+cd $TEMPORARY_REPOSITORY
+
+echo "Checking out $RELEASE_BRANCH (commit $GITHUB_SHA)"
+git checkout $GITHUB_SHA
+
+PROJECT_VERSION=$(cat gradle.properties | grep "\-SNAPSHOT" | awk -F '=' '{print $2}' | awk -F '-' '{print $1}')
```

Comment:
Hmmm this has worked for a while in the travis scripts and works here too. What's the difference in your suggestion?

Reply:
So it will technically work, but only because what is happening is this:
```
grep "\-SNAPSHOT"
```
is resolved to this ARGV array by the shell
```
{"grep", "\-SNAPSHOT"}
```
This will technically match the Gradle SNAPSHOT line because grep evaluates the regex `\-SNAPSHOT` using regex escaping rules. So `\-` will match the `-` character literally, although `\` is not necessary in this case because `-` is not a special regex character in this context. I assume you were not intending to manipulate grep's regex engine in this way.

What I imagine you intended to do was stop `grep` from treating `-SNAPSHOT` as a command option in the case like `grep -SNAPSHOT`. The correct syntax for that would be what I suggested: `grep -- -SNAPSHOT`. The `--` is a standardized option that tells a command to stop parsing options after that point.

If that was your intention, I suggested the change because it makes that intention clear to anyone reading the script.